packages/x-charts/src/BarChart/BarPlot.tsx

```diff
@@ -238,7 +238,28 @@ const BarPlotRoot = styled('g', {
   overridesResolver: (_, styles) => styles.root,
 })({
   [`& .${barElementClasses.root}`]: {
-    transition: 'opacity 0.2s ease-in, fill 0.2s ease-in',
+    [`&.${barElementClasses.vertical}`]: {
+      transition: 'opacity 0.2s ease-in, fill 0.2s ease-in, height 0.5s ease-in, y 0.5s ease-in',
+
+      '@keyframes growHeight': {
+        from: { transform: 'scaleY(0%)' },
+        to: { transform: 'scaleY(100%)' },
+      },
+
+      transformOrigin: '0% 100%',
```

Comment:
I would assume `transformOrigin` should be `50% 100%` for regular cases, while horizontal should probably be `0% 50%`. But I guess it doesn't matter much, since we don't animate "bar width" do we? 🤔

Reply:
> we don't animate "bar width" do we?

At the moment, we do animate bar width for vertical layout, but as I understood from my conversation with @alexfauquette it's by accident. We didn't intend to do it, but it was done for free with react-spring.

Reply:
I just found one edge case where applying transition to the width could be useful. But I'm not super concerned by this one. We could remove it

https://github.com/user-attachments/assets/38a922d4-9c5c-4b88-b094-f9cea9ec3020